### PR TITLE
deploy: Fix and validate executability of Jazzer jar

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

File: deploy/BUILD.bazel [hunk 1]
```diff
@@ -90,6 +90,14 @@ java_export(
     ],
 )
 
+sh_test(
```
Comment (bertschneider) [inline deploy/BUILD.bazel:93]: Shouldn't this be used somewhere? I can not find any call sites.
A local run exits with code `126` and this message (when printing it):
```
/[...]/jazzer_version_test: line 37: /[...]/execroot/jazzer/bazel-out/k8-opt/bin/deploy/jazzer_version_test.runfiles/jazzer/deploy/jazzer-project.jar: cannot execute binary file: Exec format error
```

Comment (fmeum) [reply deploy/BUILD.bazel:93]: I pushed a new version, could you test this again?

File: deploy/deploy.sh [hunk 2]
```diff
@@ -33,6 +33,9 @@ JAZZER_COORDINATES=$1
 [ ! -f "${JAZZER_JAR_PATH}" ] && \
   fail "JAZZER_JAR_PATH does not exist at '$JAZZER_JAR_PATH'"
 
+"${JAZZER_JAR_PATH}" --version 2>&1 | grep '^Jazzer v' || \
```
Comment (bertschneider) [inline deploy/deploy.sh:36]: Run this with the `java` executable as well?

Comment (fmeum) [reply deploy/deploy.sh]: Done (using the hermetic JDK for style points :-))

**Review by bertschneider — APPROVED** [review]

File: deploy/deploy.sh (hunk 2 above)
Comment (bertschneider) [reply deploy/deploy.sh]: haha, nice :smile:

**Pull request merged**